### DISCUSSION TITLE
No Github Deploy key was added while bootstraping

### DISCUSSION THREAD

**Discussion by electron2302** [fluxcd/flux2#3141] [Q&A] [answered]
Hi, I am new to this. I stumbled upon something where I guess it’s my fault or I misunderstood something but might also be a bug 

After running `flux bootstrap github  --components-extra=image-reflector-controller,image-automation-controller --owner=<githubname>   --repository=<reponame>   --branch=main   --path=clusters/main   --personal   --token-auth --read-write-key` no GitHub Deploy key was added to the repository and the “flux-system” secret was not a key but a username password combo. Since the doc for `--read-write-key ` says that “the deploy key is configured with read/write permissions” I thought a deploy key would be generated. 

Any thoughts ?

**Comment by stefanprodan** [comment, marked as answer]
When using —token-auth, Flux no longer uses SSH, so no deploy keys. If you want to switch from HTTPS and token auth to SSH and deploy keys, then delete the flux-system secret and rerun bootstrap without —token-auth.

**Reply by electron2302** [reply]
Oh yes thank you very much. 
I used that because I thought it was needed so that the cli asks me for the PAT instead of getting it over an env var.

**Reply by aabmets** [reply]
I can't run bootstrap without --token-auth, I get this error: "SSH key scan for host github.com:22 failed, error: dial tcp 140.82.121.4:22: connectex: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond." I don't understand what SSH key do I need to create. Also, how can I create the deploy key manually, because FluxCD always gives them very bad names.

**Reply by kingdonb** [reply]
@aabmets Flux names the deploy keys so they include enough relevant information that one repository can be bootstrapped from many clusters and none of the deploy keys will conflict. The connection failure could indicate a problem connecting to GitHub from in your cluster or from at your workstation. This sounds like a networking issue, either your workstation or your cluster or both cannot reach GitHub.

You can create deploy keys manually with `flux create secret git`, check out `flux create secret git --help` for more details. The secret that gets emitted on your screen is a public key, what you copy to the GitHub deploy key. The private key gets created in the cluster and is never shown. (but if the cluster cannot connect to GitHub and/or your workstation cannot connect, this may still fail – the connection to the host from your workstation is needed to get the SSH host key.)

There is also a section about bootstrapping in the end-to-end guide, so you can read about how this process works in practice and in the greater context of how Flux at-large works: https://fluxcd.io/flux/flux-e2e/#bootstrapping-flux
